src/common/rbac/project/const.go

```diff
@@ -20,14 +20,57 @@ import (
 
 // const action variables
 const (
-	ActionAll      = rbac.Action("*")
-	ActionPull     = rbac.Action("pull")
-	ActionPush     = rbac.Action("push")
-	ActionPushPull = rbac.Action("push+pull")
+	ActionAll = rbac.Action("*") // action match any other actions
+
+	ActionPull     = rbac.Action("pull")      // pull repository tag
+	ActionPush     = rbac.Action("push")      // push repository tag
+	ActionPushPull = rbac.Action("push+pull") // compatible with security all perm of project
+
+	// create, read, update, delete, list actions compatible with restful api methods
+	ActionCreate = rbac.Action("create")
+	ActionRead   = rbac.Action("read")
+	ActionUpdate = rbac.Action("update")
+	ActionDelete = rbac.Action("delete")
+	ActionList   = rbac.Action("list")
+
+	// execute replication for the replication policy (replication rule)
+	ActionExecute = rbac.Action("execute")
+
+	// vulnerabilities scan for repository tag (aka, image tag)
+	ActionScan = rbac.Action("scan")
+
+	// retag repository tag (aka, image tag) to another repository tag, user can do retag only this action allowed,
+	// retag will success only pull allowed for src repository tag and push allowed for dest repository tag
+	ActionReTag = rbac.Action("retag")
+
+	// list vulnerabilities of repository tag (aka, image tag)
+	ActionListVulnerabilities = rbac.Action("list-vulnerabilities")
+	// read manifest of repository tag (aka, image build history)
+	ActionReadManifest = rbac.Action("read-manifest")
+
+	// attach label to repository tag (aka, image tag) or helm chart version
+	ActionAddLabel = rbac.Action("add-label")
+	// detach label to repository tag (aka, image tag) or helm chart version
+	ActionRemoveLabel = rbac.Action("remove-label")
+
+	// upload helm chart or helm chart version
+	ActionUpload = rbac.Action("upload")
+	// download helm chart or helm chart version
+	ActionDownload = rbac.Action("download")
```

Comment:
Why we need so many actions? I think most of them are CRUD? 
The `ActionPull `, `ActionRead ` and `ActionDownload ` are all `Read`?
`ActionReTag ` is the combination of `Read` source `Image` and `Write` destination `Image`?

Reply:
`ActionPull` is used by docker client, `ActionDownload` now changed to `ActionRead`, `ActionReTag` is removed.